src/util/renderers/visual-editor.renderer.ts

```diff
@@ -54,19 +53,26 @@ export class VisualEditorRenderer extends Renderer {
     return ''
   }
 
-  private injectHeader(content: string, articleDetail: any): string {
-    const doc = domino.createDocument(content)
-    const header = doc.createElement('h1')
-
-    if (articleDetail?.title) {
-      header.appendChild(doc.createTextNode(articleDetail.title))
+  public async render(renderOpts: RenderOpts): Promise<any> {
+    try {
+      const result: RenderOutput = []
+      const { redisStore, articleId, articleDetail, webp, _moduleDependencies, dump } = renderOpts
+      const { html, displayTitle } = await this.getHTML(renderOpts)
+      if (html) {
+        const { finalHTML, mediaDependencies, subtitles } = await super.processHtml(html, redisStore, dump, articleId, articleDetail, _moduleDependencies, webp)
```

Comment:
`getHTML()` vs `processHTML()`, please decides one or the other form, be keep it coherent.

Reply:
Due to the logic behind `getHTML()` I incline to rename it to the `processRender()` or `preRender()`. `processHTML` will stand for treatments as before. Or are you concerned about capitalization `HTML` vs `Html`?

Reply:
I hardly can follow why a function in charge of retrieving HTML should be called `preRender()`! This is just unclear what it does. Maybe `retrieveHTML()` would be a better name? Yes I'm worry about capitalisation, make it coherent, one way or the other.

Reply:
I decided to stick to the `retrieveHtml()` here.